Development/NutriMon/src/main/java/chillingMonsters/Controllers/StockController.java

```diff
@@ -10,98 +10,98 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
-import chillingMonsters.MySQLCon;
+import chillingMonsters.DBConnect;
 
 public class StockController extends NutriMonController {
-  StockController() {
-    super("stockitems", "stockItemID");
-  }
-
-  /**
-   * Return number of servings of a recipe that can be made from user's stock.
-   *
-   * @param recipeIngredients map of the food ids and quantities that make up a recipe.
-   * @return number of servings that can be made
-   */
-  public int includesIngredients(Map<Integer, Float> recipeIngredients) {
-    Map<Integer, Float> stockIngredients = new HashMap<>();
-    String queryString = "Select foodID, sum(sQuantity) AS 'quantity' " +
-            "FROM stockitems " +
-            "WHERE userID = ? " +
-            "GROUP BY foodID";
-    try {
-      ResultSet rs;
-      try (PreparedStatement stmt = MySQLCon.getConnection()
-              .prepareStatement(queryString)) {
-        stmt.setLong(1, userId);
-        rs = stmt.executeQuery();
-      }
-      List<Map<String, Object>> stock = MySQLCon.resultsList(rs);
-      for (Map<String, Object> stockItem : stock) {
-        stockIngredients.put((Integer)stockItem.get("foodID"), (Float)stockItem.get("quantity"));
-      }
-      rs.close();
-      MySQLCon.close();
-    } catch (SQLException e) {
-      Logger.getLogger(MySQLCon.class.getName()).log(Level.SEVERE, e.getMessage(), e);
-    }
-    if (recipeIngredients.size() > stockIngredients.size()) {
-      return 0;
+    StockController() {
```

Comment:
Have you tested that these queries still work with the new connection?

Reply:
It works on login/logout, I didn't change your code. the Resultset might still be the problem

Reply:
I will fix any issues on the stock controller